htdocs/compta/paiement/class/paiement.class.php

```diff
@@ -276,12 +276,22 @@ public function create($user, $closepaidinvoices = 0, $thirdparty = null)
 				return -1;
 			}
 			if (empty($currencyofpayment)) {
-				$currencyofpayment = $this->multicurrency_code[$key];
+				if (array_key_exists($key, $this->multicurrency_code)) {
+					$currencyofpayment = $this->multicurrency_code[$key];
+				} else {
+					// Handle the case where the key does not exist in the array
+					// ...
+				}
 			}
-			if ($currencyofpayment != $this->multicurrency_code[$key]) {
-				// If we have invoices with different currencies in the payment, we stop here
-				$this->error = 'ErrorYouTryToPayInvoicesWithDifferentCurrenciesInSamePayment';
-				return -1;
+			if (array_key_exists($key, $this->multicurrency_code)) {
```

Comment:
Behaviour differs. Just replace
`if ($currencyofpayment != $this->multicurrency_code[$key]) {`
with
`if (empty($this->multicurrency_code[$key]) || $currencyofpayment != $this->multicurrency_code[$key]) {`